src/rdkafka_broker.c

```diff
@@ -1676,7 +1676,10 @@ int rd_kafka_socket_cb_generic (int domain, int type, int protocol,
         if (s == -1)
                 return -1;
 #ifdef FD_CLOEXEC
-        fcntl(s, F_SETFD, FD_CLOEXEC, &on);
+        if (fcntl(s, F_SETFD, FD_CLOEXEC, &on) == -1)
+                fprintf(stderr, "WARNING: librdkafka: %s: "
```

Comment:
can you do something else with logging here? people using a custom librdkafka log handler will not see this.

Reply:
Yeah that'd be good but unfortunately we don't have access to the rk here.
This warning is just a best effort thing to let the user know CLOEXEC failed, which should be extremely rare, if possible at all.